packages/actor-init-sparql/lib/ActorInitSparql.ts

```diff
@@ -25,57 +25,131 @@ export {
+    -d            sets a datetime for querying Memento-enabled archives
+    -p            delegates all HTTP traffic through the given proxy (e.g. http://myproxy.org/?uri=)
+    --lenient     if failing requests and parsing errors should be logged instead of causing a hard crash
+    --help        print this help message
+    --listformats prints the supported MIME types
+    --version     prints version information
+  `;
+
   public constructor(args: IActorInitSparqlArgs) {
     super(args);
   }
 
-  public async run(action: IActionInit): Promise<IActorOutputInit> {
-    const args = minimist(action.argv);
-    if (!args.listformats && (!this.queryString && (!(args.q || args.f) && args._.length < (args.c ? 1 : 2) ||
-        args._.length < (args.c ? 0 : 1) || args.h || args.help || args.v || args.version))) {
-      // Print version information
-      if (args.v || args.version) {
-        const comunicaVersion: string = require('../package.json').version;
-        const dev: string = this.isDevelopmentEnvironment() ? '(dev)' : '';
-        const nodeVersion: string = process.version;
-        const npmVersion: string = await this.getScriptOutput('npm -v', '_NPM is unavailable_');
-        const yarnVersion: string = await this.getScriptOutput('yarn -v', '_Yarn is unavailable_');
-        const os = `${OS.platform()} (${OS.type()} ${OS.release()})`;
-        return { stderr: require('streamify-string')(`| software            | version
+  public static getScriptOutput(command: string, fallback: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      exec(command, (error, stdout, stderr) => {
+        if (error) {
+          resolve(fallback);
+        }
+        resolve((stdout || stderr).trimEnd());
+      });
+    });
+  }
+
+  public static isDevelopmentEnvironment(): boolean {
+    return existsSync(`${__dirname}/../test`);
+  }
+
+  public static async buildContext(args: minimist.ParsedArgs, queryContext: boolean, helpMessage: string,
+    queryString?: string): Promise<any> {
+    // Print version information
+    if (args.v || args.version) {
+      const comunicaVersion: string = require('../package.json').version;
+      const dev: string = this.isDevelopmentEnvironment() ? '(dev)' : '';
+      const nodeVersion: string = process.version;
+      const npmVersion: string = await this.getScriptOutput('npm -v', '_NPM is unavailable_');
+      const yarnVersion: string = await this.getScriptOutput('yarn -v', '_Yarn is unavailable_');
+      const os = `${OS.platform()} (${OS.type()} ${OS.release()})`;
+
+      const message = require('streamify-string')(`| software            | version
 | ------------------- | -------
 | Comunica Init Actor | ${comunicaVersion} ${dev}
 | node                | ${nodeVersion}
 | npm                 | ${npmVersion}
 | yarn                | ${yarnVersion}
 | Operating System    | ${os}
-`) };
-      }
+`);
+
+      return Promise.reject(message);
```

Comment:
`Promise.reject` is only allowed to take an Error, so you'll have to catch the error, and pass the message and streamify it higher up.

Reply:
The code I wrote did work, the help and version messages were written on the command line. When I change it to "new Error(message)" and I print the error, there is "Error:" before the help or version message and I don't think that is wanted?